sn_cli/src/subcommands/files.rs

```diff
@@ -137,39 +137,47 @@ pub(crate) async fn files_cmds(
     Ok(())
 }
 
+/// Chunk all the files in the provided `files_path`
+/// `chunks_dir` is used to store the results of the self-encryption process
 pub(super) async fn chunk_path(
     file_api: &Files,
     files_path: &Path,
     chunks_dir: &Path,
 ) -> Result<BTreeMap<XorName, ChunkedFile>> {
     trace!("Starting to chunk {files_path:?} now.");
+    let now = Instant::now();
 
-    let total_files = WalkDir::new(files_path)
+    let files_to_chunk = WalkDir::new(files_path)
         .into_iter()
         .flatten()
-        .filter(|entry| entry.file_type().is_file())
-        .count();
+        .filter_map(|entry| {
+            if entry.file_type().is_file() {
+                if let Some(file_name) = entry.file_name().to_str() {
+                    Some((file_name.to_string(), entry.into_path()))
+                } else {
+                    println!(
+                        "Skipping file {:?} as it is not valid UTF-8.",
+                        entry.file_name()
+                    );
+                    None
+                }
+            } else {
+                None
+            }
```

Comment:
I prefer an early return for clarity and keeping indentation low. (Check syntax before incorporating.)

```suggestion
            if !entry.file_type().is_file() {
                return None
            }

            if let Some(file_name) = entry.file_name().to_str() {
                Some((file_name.to_string(), entry.into_path()))
            } else {
                println!(
                    "Skipping file {:?} as it is not valid UTF-8.",
                    entry.file_name()
                );
                None
            }
        }
```

Reply:
Thank you! Updated the PR.